src/main/java/net/sourceforge/jwbf/mediawiki/actions/queries/WatchList.java

```diff
@@ -0,0 +1,238 @@
+package net.sourceforge.jwbf.mediawiki.actions.queries;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+
+import net.sourceforge.jwbf.core.actions.RequestBuilder;
+import net.sourceforge.jwbf.core.actions.util.ActionException;
+import net.sourceforge.jwbf.core.actions.util.HttpAction;
+import net.sourceforge.jwbf.core.contentRep.WatchListResults;
+import net.sourceforge.jwbf.core.contentRep.WatchResponse;
+import net.sourceforge.jwbf.mediawiki.ApiRequestBuilder;
+import net.sourceforge.jwbf.mediawiki.MediaWiki;
+import net.sourceforge.jwbf.mediawiki.bots.MediaWikiBot;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Joiner;
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import com.google.common.primitives.Ints;
+
+public class WatchList extends BaseQuery<WatchResponse> {
+
+	public enum WatchListProperties {
+		//TODO: add all the properties
+		USER("user"), TITLE("title"), COMMENT("comment"), TIMESTAMP("timestamp"), PARSED_COMMENT(
+				"parsedcomment");
+
+		private String name;
+
+		private WatchListProperties(String name) {
+			this.name = name;
+		}
+
+		public String toString() {
+			return name;
+		}
+	}
+
+	public static final int DEFAULT_NS = 0;
+	private static final Joiner PARAM_JOINER = Joiner.on('|');
+
+	private static final Logger log = LoggerFactory.getLogger(WatchList.class);
+
+	private final ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
I used JsonMapper but you should add a method to configure it. Perhaps it will cause problems if two classes will configure it differently